app/api/services/order_service.py

```diff
@@ -0,0 +1,129 @@
+from datetime import datetime
+from uuid import uuid4, UUID
+from typing import Optional, List
+from app.api.exceptions.global_exceptions import OrderNotFoundException, ProductDoesNotExistException, OutOfStockException, StatusNotFoundException
+from app.api.services.order_status_service import OrderStatusService
+from app.api.services.product_service import ProductService 
+from app.models import Order, OrderProduct
+from decimal import Decimal
+from app.schemas.order import OrderCreationResponse, OrderItem, OrderResponse
+from fastapi import FastAPI, HTTPException, APIRouter, Query
+from sqlalchemy.orm import Session, joinedload  
+
+
+product_service = ProductService()
+order_status_service = OrderStatusService()
+
+class OrderService:
+    def __init__(self, db: Session):
+        self.db = db
+  
+    def create_order(self, order_items: List[OrderItem], user_id: Optional[UUID] = None) -> OrderCreationResponse:
+        total_price = Decimal(0)
+
+        status_id = order_status_service.get_order_status_by_name("pending").id
+        order = Order(
+            user_id=user_id,
+            total_price=total_price,
+            status_id=status_id
+        )
+        
+        self.db.add(order)
+        self.db.flush()
+
+        for index, item in enumerate(order_items):
+            product = self.product_service.get_product(item.product_id)
+            if product is None:
+                raise ProductDoesNotExistException(index + 1)
+            elif item.quantity > product.stock:
```

Comment:
First, there's no need for `elif` because you're raising an exception in the `if` statement. Also, check the availability using `product.is_available`.

Reply:
I do not recognize "product.is_available" in the code and can't find it, can you please elaborate more?

Reply:
The product has [is_available](https://github.com/AHL-Logics-backend-tranining-2024/Final-Project---Team-3/blob/ff9fdbf992d97fb89b941482aeb04254437c9501/app/models.py#L58) to determine if the user can order that product. `is_available` is different from stock; a product can be in stock but not available for a period of time.

Reply:
should be done, the model of the product had extra fields that weren't updated, now it should be good I believe